packages/edit-site/src/components/navigation-sidebar/navigation-panel/index.js

```diff
@@ -80,14 +127,67 @@ const NavigationPanel = ( { isOpen } ) => {
 				</div>
 
 				<div className="edit-site-navigation-panel__scroll-container">
-					{ contentActiveMenu === MENU_ROOT && (
-						<TemplatesNavigation />
-					) }
-					{ templatesActiveMenu === MENU_ROOT && (
-						<ContentNavigation
-							onActivateMenu={ setContentActiveMenu }
-						/>
-					) }
+					<Navigation
+						activeItem={ activeItem }
+						activeMenu={ activeMenu }
+						onActivateMenu={ ( menu ) => {
+							if ( menu === MENU_ROOT ) {
+								setNavigationPanelActiveMenu( menu );
+								setContentActiveMenu( menu );
+							} else if ( menu.startsWith( MENU_TEMPLATES ) ) {
+								setNavigationPanelActiveMenu( menu );
+							} else {
+								setContentActiveMenu( menu );
+							}
+						} }
```

Comment:
What do you think if we moved this into a proper function instead of defining it all as a prop?
It could improve the readability a little bit.

Something like this:
```jsx
const onActivateMenu = ( menu ) => { /* ... */ };

// ...

<Navigation onActivateMenu={ onActivateMenu } />
```